tests/testthat/test_plotPValHist.R

```diff
@@ -2,28 +2,77 @@ context("DGEobj.plots - tests for plotPValHist.R functions")
 
 
 test_that("plotPValHist.R: plotPvalHist()", {
-    skip_if(is.null(t_obj1$DGEList))
 
-    # testing plotPvalHist with savePlot and facet = TRUE
+    # testing plotPvalHist with facet = TRUE
     pvalMatrix <- extractCol(getType(t_obj1, "topTable"), colName = "P.Value", robust = FALSE)
     pval_plot <- plotPvalHist(pvalMatrix)
     expect_s3_class(pval_plot, c("canvasXpress", "htmlwidget"))
 
     pval_plot <- plotPvalHist(pvalMatrix, plotType = "ggplot")
     expect_s3_class(pval_plot, c("gg","ggplot"))
 
-    # testing plotPvalHist with savePlot and facet = FALSE
+    # testing plotPvalHist with facet = FALSE
     pval_plot <- plotPvalHist(as.matrix(pvalMatrix),
                               facet     = FALSE)
     expect_length(pval_plot, 4)
     expect_s3_class(pval_plot[[1]], c("canvasXpress", "htmlwidget"))
 
     pval_plot <- plotPvalHist(as.matrix(pvalMatrix),
                               plotType = "ggplot",
-                              facet     = FALSE)
+                              facet    = FALSE)
     expect_length(pval_plot, 4)
     expect_s3_class(pval_plot[[1]], c("gg","ggplot"))
 
+    #testing assert statements
+    expect_error((plotPvalHist()),
+                  reegexp = "P.Val must be specified and must be of class matrix or dataframe.")
+    expect_error(plotPvalHist(iris),
```

Comment:
here you are testing that the dataframe contains only numeric values and not discreate vs continues test, is that what need?

Reply:
You cannot bin a character variable. As long as the value is numeric , the plot works fine.